lib/determinator/tracking/rack/middleware.rb

```diff
@@ -16,7 +16,23 @@ def call(env)
           error = true
           raise
         ensure
-          Determinator::Tracking.finish!(status: status, error: !!error)
+          Determinator::Tracking.finish!(
+            status: status,
+            error: !!error,
+            endpoint: extract_endpoint(env)
+          )
+        end
+
+        private
+
+        def extract_endpoint(env)
+          if params = env['action_dispatch.request.path_parameters']
+            [params[:controller], params[:action]].join('#')
```

Comment:
What exactly does the info in `env['action_dispatch.request.path_parameters']` signify and how does it relate to HTTP requests, if we want to parse the endpoint info for http we can do
`
verb , url = endpoint.split(' ')
`
how does this fit into that scheme?

Reply:
This is added by Rails (if the request is within a Rails app). We do not intend to parse the endpoint (in fact, we're also using it for background jobs); we only want to use it to pinpoint which actions are causing problems. And if we have the Rails info, it's more informative.

Reply:
Just a thought, would having service name part of endpoint be useful?

Reply:
Hmm, that's a good point

Reply:
Added it, could you have another look? It ended up being a little bit of code as we need to keep this generic!